File: client/sockets/SocketData.py
import base64

class SocketData ():
    channel = ""
    message = ""

    def __init__(self, message, channel="$MESSAGE"):
        self.channel = channel
        self.message = message

    @staticmethod
    def parse(data):
        if isinstance(data, str):
            data = data.encode()
        ddata = base64.decodebytes(data).decode()
        channel, message = ddata.split(':', 1)
        return SocketData(message, channel)

    def serialize(self):
        return base64.b64encode((self.channel + ":" + str(self.message)).encode()).decode()

    def __str__(self):
        return self.channel + ": " + str(self.message.encode())

File: server/SocketData.java
import java.util.Base64;

public class SocketData
{
    public String message;
    public String channel;
    public Exception error;
    public SocketWorker worker;

    public SocketData(String message, String channel) {
        this.channel = channel;
        this.message = message;
    }

    public static SocketData parse(String data)
    {
        String decoded = new String(Base64.getDecoder().decode(data));
        String[] parsed = decoded.split(":", 2);
        return new SocketData(parsed[1], parsed[0]);
    }

    public String serialize()
    {
        String data = channel + ":" + message;
        return new String(Base64.getEncoder().encode(data.getBytes()));
    }

    public boolean shouldDisconnect()
    {
        return channel.equals("$DISCONNECT") && message.equals("bye");
    }

    public void attachException(Exception err)
    {
        this.error = err;
    }

    @Override
    public String toString() {
        return channel + ": " + message;
    }
}